src/traits.jl

```diff
@@ -127,3 +127,20 @@ end
 # Return the "diagonal size" of a matrix - the minimum of the two dimensions
 diagsize(A::StaticMatrix) = diagsize(Size(A))
 @pure diagsize(::Size{S}) where {S} = min(S...)
+
+
+struct Small ; end
+struct Large ; end
+
+"""
+    SizeClass(a::StaticArray, small_limit::Size)
```

Comment:
if `SizeClass` is a supertype of `Small` and `Large`, then it should be capitalized.

If it is just a function, it should be lower case.